scripts/schemas/board-schema.yml

```diff
@@ -0,0 +1,73 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Nordic Semiconductor ASA
+
+## A pykwalify schema for basic validation of the structure of a
+## board metadata YAML file.
+##
+# The board.yml file is a simple list of key value pairs containing board
+# information like: name, vendor, socs, variants.
+schema;variant-schema:
+  required: false
+  type: seq
+  sequence:
+    - type: map
+      mapping:
+        name:
+          required: true
+          type: str
+        cpucluster:
+          required: false
+          type: str
+        variants:
+          required: false
+          include: variant-schema
+
+type: map
+mapping:
+  board:
+    required: true
+    type: map
+    mapping:
+      name:
+        required: true
+        type: str
+        desc: Name of the board
+      vendor:
+        required: false
+        type: str
+        desc: SoC family of the SoC on the board.
+      revision:
+        required: false
+        type: map
+        mapping:
+          format:
+            required: true
+            type: str
+            enum:
+              ["major.minor.patch", "letter", "number", "fuzzy", "custom"]
```

Comment:
Having `fuzzy` as a format makes no sense, given how it corresponds to `board_check_revision()`:
```
fuzzy             ~ MAJOR.MINOR.PATCH
major.minor.patch ~ MAJOR.MINOR.PATCH EXACT
```
On the other hand, you can't express `[LETTER|NUMBER] EXACT` in this schema.
How about adding a separate boolean field for `exact`?

Reply:
fair point.

I actually based it on today's usage where none of the boards are using the `exact` keyword, and thus simplified to just the forms we use, but maybe allowing `exact: true` (and remove `fuzzy`) in the board schema is a better option.

Reply:
done.

`fuzzy` removed and `exact: true` supported instead.